content/en/device-provisioning-service/client-library.md

```diff
@@ -49,49 +48,146 @@ void plgd_dps_shutdown(void);
-                                     void *on_cloud_change_data);
+/**
+ * @brief Set DPS manager callbacks.
+ *
+ * @param ctx dps context (cannot be NULL)
+ * @param on_change_cb callback invoked on provisioning status change
+ * @param on_change_data user data provided to on_change_cb invocation
+ * @param on_cloud_change_cb callback invoked on cloud status change
+ * @param on_cloud_change_data user data provided to on_cloud_change_cb invocation
+ *
+ * Example of plgd_dps_on_status_change_cb_t function:
+ * @code{.c}
+ * static void
+ * on_change_cb(plgd_dps_context_t *ctx, plgd_dps_status_t status, void *on_change_data) {
+ *   printf("DPS Manager Status:\n");
+ *   if (status & PLGD_DPS_INITIALIZED) {
+ *     printf("\t-Initialized\n");
+ *   }
+ *   ...
+ * }
+ * @endcode
+ *
+ * Example of oc_cloud_cb_t function:
+ * @code{.c}
+ * static void
+ * on_cloud_change_cb(oc_cloud_context_t *ctx, oc_cloud_status_t status, void *on_cloud_change_data) {
+ *   printf("Cloud Manager Status:\n");
+ *   if (status & OC_CLOUD_REGISTERED) {
+ *     printf("\t-Registered\n");
+ *   }
+ *   ...
+ * }
+ * @endcode
+ */
+void plgd_dps_set_manager_callbacks(plgd_dps_context_t *ctx, plgd_dps_on_status_change_cb_t on_change_cb,
+                                    void *on_change_data, oc_cloud_cb_t on_cloud_change_cb, void *on_cloud_change_data);
 
 /**
- * @brief Start DPS manager to provision device by given server.
+ * @brief Start DPS manager to provision device.
  *
- * Setup context, handlers, dps server and start dps manager.
+ * Setup context, global session handlers and start DPS manager.
  *
- * @param ctx device context (cannot be NULL)
- * @param endpoint endpoint of the provisioning server (cannot be NULL)
+ * Starting DPS also starts the retry mechanism, which will remain active until the device is successfully provisioned.
+ * If a provisioning step fails, it will be tried again after a time interval. The time interval depends on the retry
+ * counter (which is incremented on each retry) and uses the following values [ 10, 20, 40, 80, 120 ] in seconds.
+ * Meaning that the first retry is scheduled after 10 seconds after a failure, the second retry after 20 seconds, etc.
+ * After the interval reaches the maximal value (120 seconds) it resets back to the first value (10 seconds).
+ *
+ * @note Before starting the DPS manager, an endpoint must be set using plgd_dps_set_endpoint. Without an endpoint set
+ * the provisioning will not start.
+ *
+ * @param ctx dps context (cannot be NULL)
  * @return 0 on success
  * @return -1 on failure
  */
-int plgd_dps_manager_start(plgd_dps_context_t *ctx, const char *endpoint);
+int plgd_dps_manager_start(plgd_dps_context_t *ctx);
 
 /**
  * @brief Stop DPS manager.
  *
- * Deregister handlers, clear context, stop manager and close connection.
+ * Deregister handlers, clear context, stop DPS manager, close connection to DPS endpoint and remove identity
```

Comment:
Does it equal to disown? Are ACLs removed? Is the device disconnected from the cloud? Is it in RFOTM state?

Reply:
I'm not sure what disown does, I think in SDK the disown method removes itself as the owner and calls a factory reset, which removes everything. 

This function removes only internal stuff (`dps_ep_session_event_handler` - connection loss to DPS endpoint; `dps_interface_event_handler` - network interface changed state to up); acls are not removed, cloud status and pstat state are not changed.

The proposal I've discussed with @jkralik in the morning changes the way we handle identity certificates. The will be modified only in the credentials handler (we'll add + tag the certificates in the handler, and in the next invocation of the handler we'll remove the tagged certs and add + tag newly received ones).